dir.targets

```diff
@@ -29,12 +29,78 @@
     </ItemGroup>
   </Target>
 
+  <!-- Corefx-specific binplacing properties -->
+  <PropertyGroup>
+    <IsRuntimeAssembly Condition="'$(IsRuntimeAssembly)'=='' AND '$(IsReferenceAssembly)' != 'true' AND '$(BinPlaceRef)' != 'true' AND '$(IsTestProject)' != 'true'">true</IsRuntimeAssembly>
+    <!-- Try to determine if this is a simple library without a ref project.
+         https://github.com/dotnet/corefx/issues/14291 is tracking cleaning this up -->
+    <IsRuntimeAndReferenceAssembly Condition="'$(IsRuntimeAndReferenceAssembly)' == '' and '$(IsRuntimeAssembly)' == 'true' and Exists('$(SourceDir)/$(MSBuildProjectName)') and !Exists('$(SourceDir)/$(MSBuildProjectName)/ref') and !$(MSBuildProjectName.StartsWith('System.Private'))">true</IsRuntimeAndReferenceAssembly>
```

Comment:
@weshaggard It's probably non-obvious from the diff: I needed to change `$(AssemblyName)` to `$(MSBuildProjectName)` in these conditions, because `$(AssemblyName)` isn't actually set until `Build.Common.targets` is imported below. It's otherwise the same, though.